File: crossecore-java/src/java_/InheritanceKind.java
package java_;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public enum InheritanceKind implements org.eclipse.emf.common.util.Enumerator{
	NONE(1, "none", "none"),
	ABSTRACT(2, "abstract", "abstract"),
	FINAL(3, "final", "final")
;	public static final int NONE_VALUE = 1;
	public static final int ABSTRACT_VALUE = 2;
	public static final int FINAL_VALUE = 3;
	
	private static final InheritanceKind[] VALUES_ARRAY =
		new InheritanceKind[] {
		NONE,
		ABSTRACT,
		FINAL
		};
	public static final List<InheritanceKind> VALUES = Collections.unmodifiableList(Arrays.asList(VALUES_ARRAY));
	
	public static InheritanceKind get(String literal) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			InheritanceKind result = VALUES_ARRAY[i];
			if (result.toString().equals(literal)) {
				return result;
			}
		}
		return null;
	}
	
	public static InheritanceKind getByName(String name) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			InheritanceKind result = VALUES_ARRAY[i];
			if (result.getName().equals(name)) {
				return result;
			}
		}
		return null;
	}
	
	public static InheritanceKind get(int value) {
		switch (value) {
			
			case NONE_VALUE: return NONE;
			
			case ABSTRACT_VALUE: return ABSTRACT;
			
			case FINAL_VALUE: return FINAL;
		}
		return null;
	}
	private final int value;
	private final String name;
	private final String literal;
	
	private InheritanceKind(int value, String name, String literal) {
		this.value = value;
		this.name = name;
		this.literal = literal;
	}

	public int getValue() {
	  return value;
	}

	public String getName() {
	  return name;
	}

	public String getLiteral() {
	  return literal;
	}

	@Override
	public String toString() {
		return literal;
	}
			
}

File: crossecore-typescript/java_/InheritanceKind.ts
    export class InheritanceKind
    {
    	
		public static readonly NONE_VALUE:number = 1;
		public static readonly ABSTRACT_VALUE:number = 2;
		public static readonly FINAL_VALUE:number = 3;
		
		public static NONE:InheritanceKind = new InheritanceKind(1, "none", "none");
		public static ABSTRACT:InheritanceKind = new InheritanceKind(2, "abstract", "abstract");
		public static FINAL:InheritanceKind = new InheritanceKind(3, "final", "final");

		private static VALUES_ARRAY:Array<InheritanceKind> = [
			InheritanceKind.NONE, 
			InheritanceKind.ABSTRACT, 
			InheritanceKind.FINAL
		];

        public static get_string(literal:string):InheritanceKind
        {
            for (let i = 0; i < this.VALUES_ARRAY.length; i++)
            {
                let result = this.VALUES_ARRAY[i];
                if (result.toString() === literal)
                {
                    return result;
                }
            }
            return null;
        }

        public static getByName(name:string):InheritanceKind
        {
	        for (let i = 0; i < this.VALUES_ARRAY.length; i++)
	        {
	            let result = this.VALUES_ARRAY[i];
	            if (result.getName()==name)
	            {
	                return result;
	            }
	        }
	        return null;
        }

        public static get_number(value:number):InheritanceKind
        {
            switch (value)
            {
			case this.NONE_VALUE: return this.NONE;
			case this.ABSTRACT_VALUE: return this.ABSTRACT;
			case this.FINAL_VALUE: return this.FINAL;
            }
            return null;
        }

	    private value:number;
	    private name:string;
	    private literal:string;

	    private constructor(value:number, name:string, literal:string)
	    {
	        this.value = value;
	        this.name = name;
	        this.literal = literal;
	    }
	
	    public getLiteral():string
	    {
	        return this.literal;
	    }
	
	    public getName():string
	    {
	        return this.name;
	    }
	
	    public getValue():number
	    {
	        return this.value;
	    }
	    
	    public toString():string
	    {
	        return this.literal;
	    }
    }
